lib/Core/Executor.cpp

```diff
@@ -3555,7 +3555,37 @@ void Executor::executeMemoryOperation(ExecutionState &state,
     if (MaxSymArraySize && mo->size >= MaxSymArraySize) {
       address = toConstant(state, address, "max-sym-array-size");
     }
-    
+    // handle concrete address case first
+    if (auto *c_address = dyn_cast<ConstantExpr>(address)) {
+      auto first_offset = mo->getOffset(c_address->getZExtValue());
```

Comment:
Due to brilliance of `auto`, I have no idea if this  is a `int`, `unsigned`, `ref<Expr>` or `ref<ConstantExpr>`

Reply:
You don't need to.